### PR TITLE
feat: [RFC] Introduce Query Params

### PR DESCRIPTION
**Pull request by harshithpabbati**
This PR introduces Query Params

Todo
* [x] Pass query, variables, operationName through URI
* [x] Parse through the URI to get query, variables

### CODE REVIEW

File: packages/graphiql/src/api/actions/sessionActions.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export enum SessionActionTypes {
   OperationSucceeded = 'OperationSucceeded',
   OperationErrored = 'OperationErrored',
   TabChanged = 'TabChanged',
+  ParametersChanged = 'ParametersChanged',
```
Comment (ncthbrt) [inline packages/graphiql/src/api/actions/sessionActions.ts:11]: Think `SearchParamatersChanged` or `QueryStringParametersChanged` would be clearer. A parameter is quite a general term.

File: packages/graphiql/src/api/hooks/useQueryParams.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      .map(function (key) {
+        return (
+          encodeURIComponent(key) + '=' + encodeURIComponent(parameters[key])
+        );
+      })
+      .join('&');
+  history.replaceState(null, null, newSearch);
+}
+
+export default function useQueryParams() {
```
Comment (ncthbrt) [inline packages/graphiql/src/api/hooks/useQueryParams.ts:19]: Think this should return the query string parameters, not the operational data. 
And @acao had the suggestion to also perhaps return a setter here as well.

File: packages/graphiql/src/api/providers/GraphiQLSessionProvider.tsx [hunk 3]
```diff
@@ -40,6 +42,11 @@ export const initialState: SessionState = {
     variables: 0,
     results: 0,
   },
+  parameters: {
```
Comment (ncthbrt) [inline packages/graphiql/src/api/providers/GraphiQLSessionProvider.tsx:45]: Could probably leave this empty.

**Comment by harshithpabbati** [comment]
@acao this is ready for review now :)

**Comment by acao** [comment]
@harshithpabbati i will pivot to this once I'm back to 2.0.0, hopefully by tomorrow

**Pull request closed without merging**